### PR TITLE
fix: add datadog service label value to chart

### PR DESCRIPTION
**Pull request by davidmdm**

### CODE REVIEW

File: chart/tests/with-datadog-service-label/values.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# The GitHub app configuration.
+# Required if githubToken is not set.
+# The GitHub application private key (PEM format)
+githubAppPrivateKey: |
+  FOOBARLOREMISPUM
+
+image:
+  tag: 0.1.2
+
+datadogServiceLabel: joy-generator-chart-test
```
Comment (alexstojda) [inline chart/tests/with-datadog-service-label/values.yaml:22]: this is a bit opinionated on having datadog. 

I would go with a more generic approach of having an `additionalLabels` or, more specifically a `podLabels` field which is a map of labels & values. This is the approach most charts use.

File: chart/templates/_helpers.tpl [hunk 2]
```diff
@@ -34,6 +34,7 @@ Create chart name and version as used by the chart label.
 Common labels
 */}}
 {{- define "joy-generator.labels" -}}
+tags.datadoghq.com/service: {{ .Values.datadogServiceLabel | default .Chart.Name }}
```
Comment (alexstojda) [inline chart/templates/_helpers.tpl:37]:
```suggestion
tags.datadoghq.com/service: {{ .Values.datadogServiceLabel | default .Release.Name }}
```

What do you think about using the release name here instead?

Comment (davidmdm) [reply chart/templates/_helpers.tpl:37]: Sounds bueno!

File: chart/templates/_helpers.tpl [hunk 3]
```diff
@@ -40,6 +40,9 @@ helm.sh/chart: {{ include "joy-generator.chart" . }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
 {{- end }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- if .Values.podLabels }}
+{{ toYaml .Values.podLabels }}
+{{- end -}}
```
Comment (alexstojda) [inline chart/templates/_helpers.tpl:45]: Check your expected file, the labels here are not added to the pod, only the meta of resources. In other words the labels were added to the deployment, but not the pod template. 

This block should go below this line https://github.com/nestoca/joy-generator/blob/fa341784e0d0b5b456a130a700f1d127b44b1f10/chart/templates/deployment.yaml#L21

**Review by alexstojda — APPROVED** [review]
Muy bien!

**Pull request merged**